### DISCUSSION TITLE
Error "code signature not valid for use"

### DISCUSSION THREAD

**Discussion by DaveEveritt** [XQuartz/XQuartz#204] [Q&A] [answered]
Forgive the intrusion, but rather than file a bug report (as this isn't a bug) I'm posting an issue here. I'm trying to run a command-line application (Astrolog, on 10.14.6 Mojave) that needs X11, so installed XQuartz with Homebrew: `brew install --cask xquartz`.

On running the application I'm getting:

```
dyld: Library not loaded: /opt/X11/lib/libX11.6.dylib
  Referenced from: /Users/MYUSER/Projects/astrolog/Astrolog-v7-x11/./ast72macx11.out
  Reason: no suitable image found.  Did find:
        /opt/X11/lib/libX11.6.dylib: code signature in (/opt/X11/lib/libX11.6.dylib) not valid for use in process using Library Validation: mapping process has no Team ID and is not a platform binary
        /opt/X11/lib/libX11.6.dylib: stat() failed with errno=1
```

**Comment by jeremyhu** [comment, marked as answer]
It looks like this application you are running is signed with Library Validation enabled.  This tells the system to only load libraries into process that are either system libraries (eg: Foundation.framework) or that are signed by the same development team as the application.

XQuartz libraries this cannot be loaded into that process.  I'd expect any application that needs to use LV and X11 to ship their own libraries to ensure they can load them.
